src/spikeinterface/postprocessing/amplitude_scalings.py

```diff
@@ -291,10 +365,325 @@ def _amplitude_scalings_chunk(segment_index, start_frame, end_frame, worker_ctx)
+    traces_with_margin: np.ndarray
+        A numpy array of shape (n_samples, n_channels) containing the traces with a margin.
+    start_frame: int
+        The start frame of the chunk for traces_with_margin.
+    end_frame: int
+        The end frame of the chunk for traces_with_margin.
+    left: int
+        The left margin of the chunk for traces_with_margin.
+    right: int
+        The right margin of the chunk for traces_with_margin.
+    nbefore: int
+        The number of samples before the spike to consider for the fit.
+    all_templates: np.ndarray
+        A numpy array of shape (n_units, n_samples, n_channels) containing the templates.
+    unit_inds_to_channel_indices: dict
+        A dictionary mapping unit indices to channel indices.
+    cut_out_before: int
+        The number of samples to cut out before the spike.
+    cut_out_after: int
+        The number of samples to cut out after the spike.
+
+    Returns
+    -------
+    np.ndarray
+        The fitted scaling factors for the colliding spikes.
+    """
+    from sklearn.linear_model import LinearRegression
+
+    # make center of the spike externally
+    sample_first_centered = np.min(collision["sample_index"]) - (start_frame - left)
+    sample_last_centered = np.max(collision["sample_index"]) - (start_frame - left)
+
+    # construct sparsity as union between units' sparsity
+    sparse_indices = np.array([], dtype="int")
+    for spike in collision:
+        sparse_indices_i = unit_inds_to_channel_indices[spike["unit_index"]]
+        sparse_indices = np.union1d(sparse_indices, sparse_indices_i)
+
+    local_waveform_start = max(0, sample_first_centered - cut_out_before)
+    local_waveform_end = min(traces_with_margin.shape[0], sample_last_centered + cut_out_after)
+    local_waveform = traces_with_margin[local_waveform_start:local_waveform_end, sparse_indices]
+
+    y = local_waveform.T.flatten()
+    X = np.zeros((len(y), len(collision)))
+    for i, spike in enumerate(collision):
+        full_template = np.zeros_like(local_waveform)
+        # center wrt cutout traces
+        sample_centered = spike["sample_index"] - (start_frame - left) - local_waveform_start
+        template = all_templates[spike["unit_index"]][:, sparse_indices]
+        template_cut = template[nbefore - cut_out_before : nbefore + cut_out_after]
+        # deal with borders
+        if sample_centered - cut_out_before < 0:
+            full_template[: sample_centered + cut_out_after] = template_cut[cut_out_before - sample_centered :]
+        elif sample_centered + cut_out_after > end_frame + right:
+            full_template[sample_centered - cut_out_before :] = template_cut[: -cut_out_after - (end_frame + right)]
+        else:
+            full_template[sample_centered - cut_out_before : sample_centered + cut_out_after] = template_cut
+        X[:, i] = full_template.T.flatten()
+
+    reg = LinearRegression().fit(X, y)
```

Comment:
Should we have `fit_intercept=False` to be sure to have no intercept when fitting.
And maybe `positive=True`also.